docs/Style.thy

```diff
@@ -13,71 +13,85 @@ theory Style
       a sub-term should not go more left than its parent term.
 
-  * Support tools like diff and grep:
+    * Support tools like diff and grep:
       diff tries to find the "function" each chunk belongs to by indentation. If we remain
       compatible, diffs are much easier to read. Pull requests mean we read a lot of diffs.
       grep and other regexp search is not needed as much as it used to, but it's still often
       useful to be able to find constants and lemmas by regexp search outside of Isabelle
 
-  * Don't fight existing auto-indent where possible:
+    * Don't fight existing auto-indent where possible:
       there can be exceptions, but each time we go against defaults, it will be annoying to follow.
 
-  * Compatibility with upstream style where it exists:
+    * Compatibility with upstream style where it exists:
       there is not much consistent Isabelle distribution style, but where there is, we should follow
-      it, so that others can read our code and we can contribute upstream
-\<close>
+      it, so that others can read our code and we can contribute upstream\<close>
 
 text \<open>
-The main rules:
-  * Indent your proofs.
+  The main rules:
+    * Indent your proofs.
 
-  * Follow naming conventions.
+    * Follow naming conventions.
 
-  * Be aware that others have to read what you write.
-\<close>
+    * Be aware that others have to read what you write.\<close>
 
 text \<open>
-To not drive proof maintainers insane:
-  * Do not use `auto` except at the end of a proof, see [1].
+  To not drive proof maintainers insane:
 
-  * Never check in proofs containing `back`.
+    * Do not use `auto` except at the end of a proof, see [1].
 
-  * Instantiate bound variables in preference to leaving schematics in a subgoal
-    (i.e. use erule_tac ... allE in preference to erule allE)
+    * Never check in proofs containing `back`.
 
-  * Explicitly name variables that the proof text refers to (e.g. with rename_tac)
+    * Instantiate bound variables in preference to leaving schematics in a subgoal
+      (i.e. use erule_tac ... allE in preference to erule allE)
 
-  * Don't mix object and meta logic in a lemma statement.
-\<close>
+    * Explicitly name variables that the proof text refers to (e.g. with rename_tac)
+
+    * Don't mix object and meta logic in a lemma statement.\<close>
+
+section \<open>Text and comments\<close>
+
+text \<open>
+  Generally, prefer @{command text} comments over (* source comments *), because the former
+  will be displayed in LaTeX and HTML output. Be aware of LaTeX pitfalls and use antiquotations
+  such as @{text some_id} or @{term "1+1"} for identifiers and terms, especially when they
+  contain underscores and other sequences that already have meaning in LaTeX.
+
+  For comments inside inner-syntax terms use the \<comment> \<open>comment symbol\<close> and to comment out parts
+  of a term the \<^cancel>\<open>cancel symbol\<close>.\<close>
```

Comment:
For comments in the middle of apply-style proofs, would you use `\<comment>` or `text`? I haven't used LaTeX/HTML rendering much, so I can't recall how proofs get rendered, but I vaguely recall that including proofs is optional. I guess Isabelle would only render `text` blocks in proofs if proof-printing is enabled?

Reply:
Well spotted, I forgot about that case. In the middle of an apply proof, I'd use `\<comment>` probably or even source comments, because usually you wouldn't read apply proofs in LaTeX